core/mapserver/mapserver.go

```diff
@@ -70,12 +70,19 @@ func (m *mapServer) signRoot(ctx context.Context) (*trillian.SignedMapRoot, erro
 	if err != nil {
 		return nil, err
 	}
+	defer func() {
+		if retErr != nil {
+			if rbErr := txn.Rollback(); rbErr != nil {
+				retErr = fmt.Errorf("%v, Rollback(): %v", retErr, rbErr)
+			}
+		}
+	}()
 	root, err := m.tree.ReadRootAt(txn, epoch)
 	if err != nil {
-		return nil, fmt.Errorf("ReadRootAt err: %v", err)
+		return nil, fmt.Errorf("ReadRootAt(%v): %v", epoch, err)
```

Comment:
nit: what do you think about changing the error text to `ReadRootAt(%v) failed: %v`. I find it helpful to search long logs for the words "fail", or "error" when something goes wrong.